### PR TITLE
PDA5 Get Task Endpoint

### PR DESCRIPTION
**Pull request by ffleandro**
* Implemented getTask on FakeEngine
* Created FakeTask using Generic Map Entity base model
* Bug corrections

### CODE REVIEW

File: src/main/java/org/entando/plugins/pda/core/model/Task.java [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-    private String description;
-    private String status;
-    private Integer priority;
-    private Boolean skipable;
-    private String processId;
-    private String processInstanceId;
-    private Map<String,String> properties;
+public interface Task {
+    @JsonIgnore
+    Integer getId();
```
Comment (werbth) [inline src/main/java/org/entando/plugins/pda/core/model/Task.java]: Will the id always be able to be represented as Integer? Don't we have string ids? How about big numbers that can't fit into an integer?

**Review by werbth — COMMENTED** [review]
@ffleandro Are we going to have unit tests for this repository?

**Comment by ffleandro** [comment]
@werbth This repository contains no actual business logic, only interfaces and base classes to be used on the engine specific implementation, so there is no need to actually have any tests as there is nothing to be tested.

Also, after your comment I moved the FakeEngine classes to the tests folder, because these are helper classes used by tests of other projects.

File: src/test/java/org/entando/plugins/pda/core/service/task/FakeTaskService.java [hunk 2]
```diff
@@ -31,38 +34,36 @@
...
+        }
+
+        throw new TaskNotFoundException();
     }
 
     private List<Task> createTasks() {
         List<Task> result = new ArrayList<>();
 
-        result.add(Task.builder()
+        result.add(FakeTask.builder()
```
Comment (joewhite101) [inline src/test/java/org/entando/plugins/pda/core/service/task/FakeTaskService.java:57]: Can we rename from `FakeTask` to `MockTask`. I think it is a really good thing to have around but probably we don't want the word "fake" running around in the codebase

File: src/test/java/org/entando/plugins/pda/core/model/FakeTask.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package org.entando.plugins.pda.core.model;
+
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Builder;
+import lombok.Singular;
+
+public class FakeTask extends Task {
```
Comment (joewhite101) [inline src/test/java/org/entando/plugins/pda/core/model/FakeTask.java:8]: Rename to MockTask (and the builder too)

File: src/main/java/org/entando/plugins/pda/core/model/Task.java [hunk 4]
```diff
@@ -1,20 +1,29 @@
...
+import org.entando.web.response.BaseMapModel;
+
+@NoArgsConstructor
+public abstract class Task extends BaseMapModel {
+
+    public Task(Map<String,Object> map) {
+        super(map);
+    }
+
+    @JsonIgnore
```
Comment (joewhite101) [inline src/main/java/org/entando/plugins/pda/core/model/Task.java:15]: Are these ignored because they'll also show up in the Map<>?

**Review by joewhite101 — CHANGES_REQUESTED** [review]
Small change in renaming mocked objects.

File: src/main/java/org/entando/plugins/pda/core/model/Task.java (hunk 4 above)
Comment (ffleandro) [reply src/main/java/org/entando/plugins/pda/core/model/Task.java:15]: Yes, they would show up duplicate, one with the method name and another with the map property name.

File: src/test/java/org/entando/plugins/pda/core/model/FakeTask.java (hunk 3 above)
Comment (ffleandro) [reply src/test/java/org/entando/plugins/pda/core/model/FakeTask.java:8]: In my opinion they should be called `Fake` due to the design pattern difference compared with `Mock`:
https://stackoverflow.com/questions/346372/whats-the-difference-between-faking-mocking-and-stubbing

In this case, the `FakeEngine` is a fully working engine, however with simplified functionalities.

File: src/test/java/org/entando/plugins/pda/core/service/task/FakeTaskService.java (hunk 2 above)
Comment (ffleandro) [reply src/test/java/org/entando/plugins/pda/core/service/task/FakeTaskService.java:57]: Please see other comment.

I've also moved all of this `Fake` classes to the tests folder to make sure we don't accidentally use it in our production code.

File: src/main/java/org/entando/plugins/pda/core/model/Task.java (hunk 4 above)
Comment (joewhite101) [reply src/main/java/org/entando/plugins/pda/core/model/Task.java:15]: 👍

**Review by joewhite101 — APPROVED** [review]

**Review by werbth — APPROVED** [review]

**Pull request merged**